samples/python/mnist/mnist_training.py

```diff
@@ -0,0 +1,142 @@
+# This code is from https://github.com/pytorch/examples/blob/master/mnist/main.py
+# with modification to do training using onnxruntime as backend on cuda device.
+
+import argparse
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torchvision import datasets, transforms
+
+import onnxruntime
+from onnxruntime.experimental import ORTTrainer, ORTTrainerOptions, optim
+
+
+# Pytorch model
+class NeuralNet(nn.Module):
+    def __init__(self, input_size, hidden_size, num_classes):
+        super(NeuralNet, self).__init__()
+        self.fc1 = nn.Linear(input_size, hidden_size)
+        self.relu = nn.ReLU()
+        self.fc2 = nn.Linear(hidden_size, num_classes)
+
+    def forward(self, input1):
+        out = self.fc1(input1)
+        out = self.relu(out)
+        out = self.fc2(out)
+        return out
+
+
+# ONNX Runtime training
+def mnist_model_description():
+    return {'inputs': [('input1', ['batch', 784]),
+                       ('label', ['batch'])],
+            'outputs': [('loss', [], True),
+                        ('probability', ['batch', 10])]}
+
+
+def my_loss(x, target):
+    return F.nll_loss(F.log_softmax(x, dim=1), target)
+
+
+# Helpers
+def train_with_trainer(log_interval, trainer, device, train_loader, epoch):
+    for batch_idx, (data, target) in enumerate(train_loader):
+        # Fetch data
+        data, target = data.to(device), target.to(device)
+        data = data.reshape(data.shape[0], -1)
+
+        # Train step
+        loss, _ = trainer.train_step(data, target)
+
+        # Stats
+        if batch_idx % log_interval == 0:
+            print('Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}'.format(
+                epoch, batch_idx * len(data), len(train_loader.dataset),
+                100. * batch_idx / len(train_loader), loss))
+
+
+def test_with_trainer(trainer, device, test_loader):
+    test_loss = 0
+    correct = 0
+    with torch.no_grad():
+        for data, target in test_loader:
+            # Fetch data
+            data, target = data.to(device), target.to(device)
+            data = data.reshape(data.shape[0], -1)
+
+            # Eval step
+            # Using fetches around without eval_step to not pass 'target' as input
+            trainer._train_step_info.fetches = ['probability']
+            output = F.log_softmax(trainer.eval_step(data), dim=1)
```

Comment:
what happens if target is still passed in as an argument, would it error out?

Reply:
No, it simply doesn't use it. and run eval_step the same way as if target wasnt specified